src/EnergyPlus/WeatherManager.cc

```diff
@@ -2334,8 +2330,7 @@ namespace WeatherManager {
         state.dataEnvrn->OpaqueCloudCover = state.dataWeatherManager->TodayOpaqueSkyCover(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay);
 
         if (state.dataWeatherManager->UseRainValues) {
-            state.dataEnvrn->IsRain = state.dataWeatherManager->TodayIsRain(
-                state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay); //.or. LiquidPrecipitation >= .8d0)  ! > .8 mm
+            state.dataEnvrn->IsRain = state.dataWaterData->RainFall.CurrentAmount >= 0.8 / 1000.0 / double(state.dataGlobal->NumOfTimeStepInHour);
```

Comment:
The last part of this is a fixed value once `NumOfTimeStepInHour` is known. There should be some place in WeatherManager with one-time inits, so this could be calculated once there and stored.

Also, based on the CI diffs, this is still overriding the SizingPeriod:DesignDay input for Rain. I guess I'm ok with that. Just be sure to explain that in the docs for Site:Precipitation.

Reply:
Right right, the threshold for the rain flag should be stored.

I'll add documentation about the rain flag computed here overwriting the design day rain flag.

Reply:
Also, you may need to add some explanation about the default precip amount for the weather file and how all these parts interact.

Reply:
I'll add some more explanation about the epw precipitation, user input site precipitation, default amount, and how they collectively affect the rain flag.

Reply:
Hi @mjwitte, I felt maybe I should still keep the SizingPeriod:DesignDay rain indicator intact? As currently epw rainfall also controls the `IsRain` calculation (since we're using the `state.dataWaterData->RainFall.CurrentAmount`, which could come from both site:precipitation and epw), the inaccuracy in epw rain data might throw off the simulation? So I a check to make the overwritten of `IsRain` only happen during the actual simulation run periods.